docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+- Ingest metrics using the https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-prometheus.html[Metricbeat
+Prometheus Module] and view your metrics in {kib}.
+- Instrument your application using the https://www.elastic.co/guide/en/apm/agent/java/current/[Elastic APM
+Java agent].
+- Monitor your services using {heartbeat} and view your uptime data in {kib}.
+
+[discrete]
+=== Before you begin
+
+Create a deployment using our hosted {ess} on https://www.elastic.co/cloud/[Elastic Cloud]. The deployment includes
+an {es} cluster for storing and searching your data, {kib} for visualizing and managing
+your data, and an APM server. For more information, see <<set-up-on-cloud,Set up on Cloud>>.
+
+[discrete]
+=== Step 1: Create a Java application
+
+To create the Java application, you require OpenJDK 14 (or higher) and the https://javalin.io/[Javalin]
+web framework. The application will include the main endpoint, an
+artificially long-running endpoint, and an endpoint that needs to poll
+another data source. There will also be a background job
+running. 
+
+. Set up a Gradle project and create the following `build.gradle` file.
++
+[source,gradle]
+----
+plugins {
+  id 'java'
+  id 'application'
+}
+
+repositories {
+  jcenter()
+}
+
+dependencies {
+  implementation 'io.javalin:javalin:3.9.1'
+
+  testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
+  testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.6.2'
+}
+
+application {
+  mainClassName = 'de.spinscale.javalin.App'
+}
+
+test {
+  useJUnitPlatform()
+}
+----
++
+. Run the following command.
++
+[source,bash]
+----
+echo "rootProject.name = 'javalin-app'" >> settings.gradle
+
+mkdir -p src/main/java/de/spinscale/javalin
+mkdir -p src/test/java/de/spinscale/javalin
+----
++
+. Install the Gradle wrapper. An easy way to install Gradle is to use
+https://sdkman.io/[sdkman] and run `sdk install gradle 6.5.1`.
++
+. Run `./gradlew clean check`. You should see a
+successful build that has nothing built or compiled yet.
++
+. To create a Javalin server and its first endpoint (the main endpoint), create the
+`src/main/java/de/spinscale/javalin/App.java` file.
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.javalin.Javalin;
+
+public class App {
+    public static void main(String[] args) {
+        Javalin app = Javalin.create().start(7000);
+        app.get("/", ctx -> ctx.result("Appsolutely perfect"));
+    }
+}
+----
++
+. Run `./gradlew assemble`
++
+This command compiled the `App.class` file in the `build` directory.
+However, there is no way to start the server. Let’s create a jar which
+contains our compiled class along with all the required dependencies.
++
+. In the `build.gradle` file, edit `plugins` as shown here.
++
+[source,gradle]
+----
+plugins {
+  id 'com.github.johnrengelman.shadow' version '6.0.0'
+  id 'java'
+}
+----
++
+. Run `./gradlew shadowJar`. This command creates a
+`build/libs/javalin-app-all.jar` file.
```

Comment:
I can't get past this step. I get the following error:

```
Brandons-MacBook-Pro-4 elastic-obs-java-app $ ./gradlew shadowJar

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/brandonmorelli/Documents/GitHub/elastic-obs-java-app/build.gradle' line: 17

* What went wrong:
A problem occurred evaluating root project 'javalin-app'.
> Could not find method application() for arguments [build_954lfwy4urhht8sn6t17frb29$_run_closure3@71d3bfa7] on root project 'javalin-app' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

Reply:
can you run `./gradlew --version` and paste the results here?

Reply:
These are the results when I run `./gradlew --version`

------------------------------------------------------------
Gradle 6.5.1
------------------------------------------------------------

Build time:   2020-06-30 06:32:47 UTC
Revision:     66bc713f7169626a7f0134bf452abde51550ea0a

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          14.0.2 (Oracle Corporation 14.0.2+12-46)
OS:           Mac OS X 10.14.5 x86_64

Reply:
```
------------------------------------------------------------
Gradle 6.5.1
------------------------------------------------------------

Build time:   2020-06-30 06:32:47 UTC
Revision:     66bc713f7169626a7f0134bf452abde51550ea0a

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          14.0.2 (Oracle Corporation 14.0.2+12-46)
OS:           Mac OS X 10.15.6 x86_64
```
Looks to be the same as Eamonn

Reply:
That looks good. Can you share your project somewhere so I can take a look?